src/cmap/connection_pool.ts

```diff
@@ -463,14 +488,20 @@ function processWaitQueue(pool: any) {
     }
 
     const connection = pool[kConnections].shift();
+    if (!connection) {
```

Comment:
similar bug here, but more subtle. If there is no connection then we want to `break` and process the wait queue later when the driver determines there are more connections